### ISSUE TITLE
Cannot read property "realFetch" of undefined

### ISSUE THREAD

**Issue by zedd45** [wheresrhys/fetch-mock#106]
I'm using [`karma-webpack`](https://github.com/webpack/karma-webpack) `v1.7.0`, and I'm trying to write tests for my [Redux action creators](http://redux.js.org/docs/recipes/WritingTests.html#action-creators).   The problem i'm having is that the `this` context is `undefined` in my tests when `fetch-mock` tries to [look that property up](https://github.com/wheresrhys/fetch-mock/blob/e699a9377cc79c12c652d0c454ee62d4b05f26b7/src/fetch-mock.js#L195). 

<img width="344" alt="realfetch undefined" src="https://cloud.githubusercontent.com/assets/546437/16395461/13e83aea-3c88-11e6-9031-700517ca2edc.png">

Additionally (but clearly not the root issue):
I can tell that fetch is defined, and global, from the karma debug window / Chrome console: 
``` 
> window.fetch
fetch() { [native code] }
```

Here's a gist of my test code: 
https://gist.github.com/zedd45/5600828f75ecee2e0a84570bd9f9d848

I'm not sure what's changing my context. Any ideas?

**Comment by wheresrhys** [comment]
My guess is you can't assign `mock = fetchMock.mock` - the `mock` method modifies the `fetchMock` object. You could do `mock = fetchMock.mock.bind(fetchMock)` ... though why not just use `fetchMock.mock` everywhere. Makes for more readable code IMHO

**Comment by zedd45** [comment]
I'll give the later a shot, first.  Maybe just an oversight on my part 😄

**Comment by zedd45** [comment]
That seems to have fixed the problem.  My matcher needs some adjustment, as I'm getting an http error like when `node-nock` fails, but that is almost certainly operator error, as well. 

```
Error: request failed
    at handleHTTPError
```

Thanks for your help!

**Comment by zedd45** [comment]
For posterity, I did something else ignorant: I shadowed the scope of `fetch` in my Action Creator by importing fetch as a local variable.  I've since removed `import fetch from 'isomorphic-fetch;` and allowed the globals to assign properly, and it works.  (Aka don't structure your code like the [async example](https://github.com/reactjs/redux/blob/master/examples/async/actions/index.js#L1) in `redux`, because this will create a [variable shadow](https://en.wikipedia.org/wiki/Variable_shadowing) that will interfere with this library.
